homework3/semi-uber/Server/api/handlers/userHandlers/userDeteleHandler.js

```diff
@@ -2,25 +2,39 @@ const log4js = require("log4js");
 const logger = log4js.getLogger();
 const { User } = require("../../Schemas/user.schema");
 const { Load } = require("../../Schemas/load.schema");
+const { Truck } = require("../../Schemas/truck.schema");
 const errorHandler = require("../error.handler");
+const checkForEdit = require("../../utilits/checkforEdit");
 
-function userDeleteHandler(req, res) {
+async function userDeleteHandler(req, res) {
   const { id: userID } = req.params;
 
-  User.findByIdAndDelete(userID)
-    .then(() => Load.remove({ created_by: userID }))
-    .then(() => {
-      logger.info(
-        "User's account and all custom data was successfully deleted"
-      );
-      res.json({
-        success: true,
-        data: {},
-        message: "User's account all custom data was successfully deleted"
-      });
-    })
-    .catch(error => {
-      return errorHandler(error.message, res);
-    });
+  const editCheck = await checkForEdit(userID);
```

Comment:
What is this check for? Does not every user has a possibility to delete his profile?

Reply:
User can not edit his profile when:

For driver:

- some of trucks is on status "OL",
For shipper:

- some of his loads on status "ASSIGNED"